src/main/java/com/github/kobeay/login_server/service/AuthService.java

```diff
@@ -1,8 +1,40 @@
 package com.github.kobeay.login_server.service;
 
+import com.github.kobeay.login_server.constant.EntityStatus;
+import com.github.kobeay.login_server.dto.RegisterRequest;
+import com.github.kobeay.login_server.dto.RegisterResponse;
+import com.github.kobeay.login_server.entity.User;
+import com.github.kobeay.login_server.repository.UserRepository;
+import com.github.kobeay.login_server.validator.JwtValidator;
+import jakarta.transaction.Transactional;
+import org.springframework.security.crypto.password.PasswordEncoder;
 import org.springframework.stereotype.Service;
 
 @Service
 public class AuthService {
+    private final UserRepository userRepository;
+    private final JwtValidator jwtValidator;
+    private final PasswordEncoder passwordEncoder;
 
+    public AuthService(UserRepository userRepository, JwtValidator jwtValidator, PasswordEncoder passwordEncoder) {
+        this.userRepository = userRepository;
+        this.jwtValidator = jwtValidator;
+        this.passwordEncoder = passwordEncoder;
+    }
+
+    @Transactional
+    public RegisterResponse register(RegisterRequest registerRequest) {
+        String encodedPassword = passwordEncoder.encode(registerRequest.getPassword());
+        User user = new User(registerRequest.getUsername(), encodedPassword, EntityStatus.ACTIVE);
+        userRepository.save(user);
+        return RegisterResponse.builder()
+                .id(user.getId())
```

Comment:
userRepository.save 함수가 실행되기 전에는 user.getId의 값이 항상 null일 것 같네요! id 값은 데이터베이스가 정해주는 key라서 다음과 같이 바꾸면 어떨까요?

```suggestion
        User user = new User(registerRequest.getUsername(), encodedPassword, EntityStatus.ACTIVE);
        User newUser = userRepository.save(user);
        return RegisterResponse.builder()
                .id(newUser.getId())
                
         //이하 모든 라인에서 user 대신 newUser 사용     
```